observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerDeploymentProcessor.java

```diff
@@ -44,7 +44,6 @@ public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws Deploym
                 deploymentPhaseContext,
```

Comment:
This service does not look particularly complicated.  Why not create a ServiceInstaller from a Supplier<MetricRegistration>?

Reply:
Something like 
```
ServiceInstaller.builder((Supplier<MicrometerDeploymentService>) () -> {
    return ...; 
});
```?